### PR TITLE
Add more setting for text styling

### LINKED ISSUE #21: [Enchancement] Add more setting for text styling
### **Is your enhancement request related to a problem? Please describe.**
The user couldn't style the text further like adding underline to the text, change to use different font family, add borders to the text.

### **Describe the enhancement you'd like**
It would be great to have a section to further change the style of the text.

### **Possible Implementation Details**
Add variables to let text show different font family, border style, text decoration. 
If user are allow to add more than 1 text i will look into how to store different style for different text.

### **Optional Sections (if relevant):**

- **Priority (Low / Medium / High)**  High
- **Are you willing to submit a PR for this enhancement?**  Yes
- **Does this enhancement require changes in documentation?** Yes

### PR DESCRIPTION
**Pull request by Laikokwui**
### **Description**
This pull request will add more text styling feature such as text formatting, change font family

### **Related Issue**

Closes #21 

### **Changes Made**
- move the text setting to popover.
- added fontfamily selector
- added text formating such as bold, underline and italic

### **Type of Change**
_Select the type of change made with this pull request:_

- [ ] New feature (non-breaking change that adds functionality)

### **Screenshots (if applicable)**
image

### **Checklist**
_Check off the following items before submitting the pull request:_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### **Additional Context**
This is not done yet. i submit first to show my progress related to issue number 21.
add text border (in progress)
fix font family setting buggy ( in progress)

I ran into 1 issue, the text positioning doesnt work anymore. i need help for this part.

### CODE REVIEW

**Comment by Laikokwui** [comment]
@suryanshsingh2001 could you help me see why the text position doesnt work?

**Comment by suryanshsingh2001** [comment]
> @suryanshsingh2001 could you help me see why the text position doesnt work?

@Laikokwui I'll have a look at this and will let you know how to proceed next.

**Comment by suryanshsingh2001** [comment]
@Laikokwui First things first, please make sure to take the latest changes from `master` branch and then add your current changes

**Comment by Laikokwui** [comment]
@suryanshsingh2001 i found the problem dy and it fixed it "textPosition.x + metrics.width" i removed metrics.width and caused it to cannot move the text

**Comment by Laikokwui** [comment]
@suryanshsingh2001 i done with the changes and also took the latest changes before add my changes. everything is working. please give me a review

**Comment by suryanshsingh2001** [comment]
> @suryanshsingh2001 i done with the changes and also took the latest changes before add my changes. everything is working. please give me a review

Hey @Laikokwui , everything looks perfect except a few changes still required

- [ ] Change the default letter spacing to 0.
- [ ] If possible add some hover labels on Toggle buttons for better accessiblity
- [ ] Add settings button below text box for better visual look, right now it looks a bit odd.

**Comment by Laikokwui** [comment]
@suryanshsingh2001 alright.

File: components/text-manager.tsx [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+            <SelectTrigger id="font-family" className="col-span-3 flex gap-2 justify-start">
+              <SelectValue placeholder="Select font family" />
+            </SelectTrigger>
+            <SelectContent>
+              <SelectItem value="Arial">Arial</SelectItem>
+              <SelectItem value="Impact">Impact</SelectItem>
+              <SelectItem value="Georgia">Georgia</SelectItem>
+              <SelectItem value="Trebuchet MS">Trebuchet MS</SelectItem>
+              <SelectItem value="Courier New">Courier New</SelectItem>
+              <SelectItem value="Comic Sans MS">Comic Sans MS</SelectItem>
```
Comment (suryanshsingh2001) [inline components/text-manager.tsx:152]: Suggestion: **Refactor** fonts into constants.ts file and then map it here

**Comment by Laikokwui** [comment]
image
@suryanshsingh2001 is this ok?

**Comment by Laikokwui** [comment]
@suryanshsingh2001 where should i store the constant.ts file?

**Comment by suryanshsingh2001** [comment]
> image @suryanshsingh2001 is this ok?

Hey @Laikokwui, is the canvas height also changing with it?

**Comment by suryanshsingh2001** [comment]
> @suryanshsingh2001 where should i store the constant.ts file?

You can store in `lib/contants.ts`

**Comment by Laikokwui** [comment]
the text settings stay the same after changing the canvas size

**Comment by suryanshsingh2001** [comment]
> the text settings stay the same after changing the canvas size

okay no issues, please proceed

**Comment by suryanshsingh2001** [comment]
> > the text settings stay the same after changing the canvas size
> 
> I mean when you are opening the Text tab, do you see canvas height being altered?


https://github.com/user-attachments/assets/14547401-ba3d-428c-be9d-98042695e1f4

**Comment by Laikokwui** [comment]
@suryanshsingh2001 oh i see, for my side no. i test changing the setting on the design and text tab and switch between them. the canvas height stay the same

**Comment by Laikokwui** [comment]
@suryanshsingh2001 done, please review and if no problem before merge please add hacktoberfest tag. thank you

**Review by suryanshsingh2001 — APPROVED** [review]
Awesome Work 👍

**Comment by suryanshsingh2001** [comment]
### 🙏 Thank You! 

We really appreciate your effort and enthusiasm in helping out. Every contribution, big or small, makes a difference! 🚀  
If you have any questions or need assistance, don’t hesitate to reach out. We're here to help! 😊

**Comment by suryanshsingh2001** [comment]
### Star the repository 🌟

Hey , if you find Mockly useful, please consider giving the repository a ⭐ on GitHub. Your support helps the project grow and encourages more contributions! 

Thank You.

**Pull request merged**